### ISSUE TITLE
Placeholder inherits NOT from nearest ancestor because of wrong order in list

### ISSUE THREAD

**Issue by SachaMPS** [divio/django-cms#4305]
**Environment**
Django==1.6.11
django-cms==3.1.1

**Issue Description**
On a page which is nested 4 levels deep in the page tree we discovered that the placeholder in the page is not inheriting from its nearest ancestor. Instead of inheriting from the nearest ancestor it inherits from the page which is the nearest to the root and contains such a plaeholder. 

**Possible Section of Code which causes this**
https://github.com/divio/django-cms/blob/develop/cms/templatetags/cms_tags.py#L219-L220

```
pages = chain([current_page], current_page.get_cached_ancestors())
````
pages is an array which contains the pages in the following order for traversal:
[current_page, root, page_level_1, page_level_2]
so what happens is that it starts with the current  page, then jumps to the root, then to the page nested one level deep, and  so on... As far as i understand it should instead go in this order: current_page, page_level_2, page_level_1 and so on.

I hope i could explain it good enough :) Thanks for all your hard work!

**Comment by yakky** [comment]
@SachaMPS thanks for reporting this, could you test #4363?

**Comment by yakky** [comment]
Fixed by #4363
